hyperspy/drawing/utils.py

```diff
@@ -1253,9 +1289,9 @@ def transparent_single_color_cmap(color):
                 foundim = i
 
         if foundim is not None:
-            f.subplots_adjust(right=0.8)
-            cbar_ax = f.add_axes([0.9, 0.1, 0.03, 0.8])
-            f.colorbar(ax_im_list[foundim], cax=cbar_ax)
+            fig.subplots_adjust(right=0.8)
+            cbar_ax = fig.add_axes([0.9, 0.1, 0.03, 0.8])
+            fig.colorbar(ax_im_list[foundim], cax=cbar_ax)
             if tight_layout:
                 # tight_layout, leaving room for the colorbar
                 plt.tight_layout(rect=[0, 0, 0.9, 1])
```

Comment:
Should this be `fig.set_tight_layout(True)` 

Per matplotlib documentation:

"""Note that [matplotlib.pyplot.tight_layout()](https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.tight_layout.html#matplotlib.pyplot.tight_layout) will only adjust the subplot params when it is called. In order to perform this adjustment each time the figure is redrawn, you can call fig.set_tight_layout(True), or, equivalently, set [rcParams["figure.autolayout"]](https://matplotlib.org/stable/users/explain/customizing.html?highlight=figure.autolayout#matplotlibrc-sample) (default: False) to True."""

Reply:
`fig.tight_layout` and `plt.tight_layout` should be doing the same in this case, with the preference for the first one, since it is more explicit and we have the figure handle.
There is a difference between `fig.tight_layout` and `fig.set_tight_layout`. and the latter is deprecated: https://matplotlib.org/stable/api/_as_gen/matplotlib.figure.Figure.set_tight_layout.html but [`fig.set_layout_engine`](https://matplotlib.org/stable/api/_as_gen/matplotlib.figure.Figure.set_layout_engine.html#matplotlib.figure.Figure.set_layout_engine) can be used instead.

For now, I would suggest to simply change to `fig.tight_layout` and leave the layout improvement to another PR, because it would be good to consider using constrained layout, as it may play better with colorbar, etc.

Reply:
@ericpre Perfect!